Packs/RSANetWitness_v11_5/Integrations/RSANetWitness115/RSANetWitness115.py

```diff
@@ -0,0 +1,1120 @@
+                                      error_handler=exception_handler)
+
+        return response
+
+    def snapshot_details_get_request(self, agent_id, snapshot_timestamp, serviceid, categories):
+        params = assign_params(serviceId=serviceid, categories=categories)
+        headers = self._headers
+
+        response = self._http_request(
+            'GET', f'rest/api/host/{agent_id}/snapshots/{snapshot_timestamp}', params=params, headers=headers,
+            error_handler=exception_handler)
+
+        return response
+
+    def files_list_request(self, serviceid, page_number, page_size):
+        params = assign_params(serviceId=serviceid, pageNumber=page_number, pageSize=page_size)
+        headers = self._headers
+
+        response = self._http_request('GET', 'rest/api/files', params=params, headers=headers,
+                                      error_handler=exception_handler)
+
+        return response
+
+    def scan_request_request(self, agent_id, serviceid, scantype, cpumax):
+        params = assign_params(serviceId=serviceid, scanType=scantype, cpuMax=cpumax)
+        headers = self._headers
+
+        response = self._http_request('POST', f'rest/api/host/{agent_id}/scan', params=params, headers=headers,
+                                      empty_valid_codes=[200], return_empty_response=True,
+                                      error_handler=exception_handler)
+
+        return response
+
+    def scan_stop_request_request(self, agent_id, serviceid, scantype):
+        params = assign_params(serviceId=serviceid, scanType=scantype)
+        headers = self._headers
+
+        response = self._http_request('DELETE', f'rest/api/host/{agent_id}/scan', params=params, headers=headers,
+                                      empty_valid_codes=[200], return_empty_response=True,
+                                      error_handler=exception_handler)
+
+        return response
+
+    def host_alerts_list_request(self, agent_id, serviceid, alertcategory):
+        params = assign_params(serviceId=serviceid, alertCategory=alertcategory)
+        headers = self._headers
+
+        response = self._http_request('GET', f'rest/api/host/{agent_id}/alerts', params=params, headers=headers,
+                                      error_handler=exception_handler)
+
+        return response
+
+    def file_alerts_list_request(self, checksum, serviceid, alertcategory):
+        params = assign_params(serviceId=serviceid, alertCategory=alertcategory)
+        headers = self._headers
+
+        response = self._http_request('GET', f'rest/api/file/{checksum}/alerts', params=params, headers=headers,
+                                      error_handler=exception_handler)
+
+        return response
+
+    def file_download_wildcard_request(self, agent_id, serviceid, path, countfiles, maxfilesize):
+        params = assign_params(serviceId=serviceid)
+        data = {"countFiles": countfiles, "maxFileSize": maxfilesize, "path": path}
```

Comment:
Is it ok to send an empty data here? Why not use the assign_params here too?
```suggestion
        data = {"countFiles": countfiles, "maxFileSize": maxfilesize, "path": path}
```

Reply:
All these parameters have to show in the data, otherwise, we get an error from API.
I think this makes it clearer.